src/main/java/org/openstreetmap/atlas/geography/boundary/CountryBoundaryMap.java

```diff
@@ -1300,230 +1425,111 @@ private Geometry nearestNeighbour(final Envelope envelope, final Object object,
+                        try
                         {
-                            final com.vividsolutions.jts.geom.Geometry geometry = reader
-                                    .read(gridIndexGeometryTokenizer.nextToken());
-                            gridIndex.insert(geometry.getEnvelopeInternal(), geometry);
+                            writer.write(GRID_INDEX_DELIMITER);
+                            writer.write(WKT_WRITER
+                                    .write(DynamicGridIndexBuilder.buildGeoBox(cell.getMinX(),
+                                            cell.getMaxX(), cell.getMinY(), cell.getMaxY())));
+                        }
+                        catch (final Exception e)
+                        {
+                            logger.error(
+                                    String.format("Failed to write cell {} for {}", cell, country));
                         }
 
-                        final com.vividsolutions.jts.geom.Geometry envelope = reader
-                                .read(gridIndexEnvelope);
-                        this.gridIndexEnvelope = envelope.getEnvelopeInternal();
-
-                        gridIndex.build();
-                        this.gridIndex = gridIndex;
-                    }
-                    catch (final Exception e)
-                    {
-                        logger.error("Invalid grid index text file format.", e);
-                        e.printStackTrace();
-                    }
-
-                    logger.info("Successfully read grid index of size {} from file.",
-                            this.gridIndex.size());
-
-                    // We can break here since the grid index is the last entry in the text file
-                    break;
-                }
+                    });
 
-                // Read the country boundaries
-                final StringTokenizer boundaryTokenizer = new StringTokenizer(line,
-                        COUNTRY_BOUNDARY_DELIMITER);
-                final String countryISO = boundaryTokenizer.nextToken();
-                final String wkt = boundaryTokenizer.nextToken();
-                try
-                {
-                    final StringTokenizer wktTokenizer = new StringTokenizer(wkt, LIST_SEPARATOR);
-                    while (wktTokenizer.hasMoreTokens())
-                    {
-                        final com.vividsolutions.jts.geom.MultiPolygon multiPolygon = (com.vividsolutions.jts.geom.MultiPolygon) reader
-                                .read(wktTokenizer.nextToken());
-                        addCountry(countryISO, multiPolygon);
-                    }
+                    writer.write(System.lineSeparator());
                 }
                 catch (final Exception e)
                 {
-                    logger.error("Invalid country boundary text file format.", e);
-                    e.printStackTrace();
+                    logger.error(String.format("Failed to write cells for {}.", country));
```

Comment:
How much value is a grid index without data for a whole country or multiple countries? From above code, it doesn't look like we'll generate the missing portions, will be fall back on the raw index at that point and slow down overall processing? If that's the case, it might not make sense to generate a partially complete grid index here. Thoughts on throwing instead of logging?

Reply:
Throwing here as well sounds good.